content/utilities/PlanetOverview/plugin.js

```diff
@@ -0,0 +1,301 @@
+    // default static text header of table
+    this.thead = document.createElement("thead");
+    this.thead.innerHTML = `<tr><th></th><th>Name</th><th>Level</th><th>Silver</th><th>SilverTime</th><th>Energy</th><th>EnergyTime</th>`;
+    this.table.appendChild(this.thead);
+    // default dynamic table body
+    this.tbody = document.createElement("tbody");
+    this.table.appendChild(this.tbody);
+  }
+  // Function to renderPlanets with sorting from home planet of preselected planet
+  renderPlanets(minLevel, planetType, middle, MyPlanets) {
+    // clean table body
+    this.tbody.innerHTML = null;
+    const selectedPlanetCoords = ui.selectedPlanet?.location?.coords;
+    const knownPlanets = [];
+    for (const planet of df.getAllPlanets()) {
+      if (planet.planetType != this.planetType) continue;
+      if (planet.planetLevel < this.minLevel) continue;
+      if (!planet?.location?.coords) continue;
+      if (myPlanets.checked && planet.owner != df.account) continue;
+
+      if (!selectedPlanetCoords) {
+        middle = df.getHomeCoords();
+      } else {
+        middle = selectedPlanetCoords;
+      }
+
+      knownPlanets.push({
+        locationId: planet.locationId,
+        distance: Math.floor(df.getDistCoords(planet.location.coords, middle)),
+      });
+    }
+
+    const sortedPlanets = knownPlanets
+      .sort((a, b) => a.distance - b.distance)
+      .slice(0, this.topX);
+    for (const [idx, p] of sortedPlanets.entries()) {
+      const planet = df.getPlanetWithId(p.locationId);
+      if (!planet) {
+        console.log(`Where is planet: ${p.locationId}`);
+        continue;
+      }
+      const row = document.createElement("tr");
+      row.style.color = getOwnerColor(planet);
+      row.onclick = () => {
+        ui.centerLocationId(planet.locationId);
+      };
+      // Create a Row per one planet
+      const silverPercent = Math.round(
+        100 / (planet.silverCap / Math.round(getSilver(planet)))
+      );
+      const fullSilverDate = new Date(
+        df.getSilverCurveAtPercent(planet, 99) * 1000
+      ).toString();
+      const fullSilverTime = [
+        fullSilverDate.substr(4, 6),
+        fullSilverDate.substr(16, 8),
+      ];
+
+      const energyPercent = Math.round(
+        100 / (planet.energyCap / Math.round(planet.energy))
+      );
+      const fullEnergyDate = new Date(
+        df.getEnergyCurveAtPercent(planet, 99) * 1000
+      ).toString();
+      const fullEnergyTime = [
+        fullEnergyDate.substr(4, 6),
+        fullEnergyDate.substr(16, 8),
+      ];
+
+      row.innerHTML = `<td>${idx + 1}.</td><td>${getPlanetName(
+        planet
+      )}</td><td style="text-align: center">${
+        planet.planetLevel
+      }</td><td>${getSilver(planet)}/${
+        planet.silverCap
+      }=${silverPercent}%</td><td>${fullSilverTime}</td><td>${Math.round(
+        planet.energy
+      )}/${planet.energyCap}=${energyPercent}%</td><td>${fullEnergyTime}</td>`;
```

Comment:
the numbers planet.energy / silver / energyCap / silverCap can get very big for big planets
to make it easier to see and using less space you could use these functions i made if you want:
```
function roundToDecimal(num, decimalCount=1) {
	if (decimalCount < 1) return Math.round(num);
	let p = Math.pow(10, decimalCount);
	num = num * p;
	num = Math.round(num) / p;
	return num;
}

function formatNumberForDisplay(num, decimalCount=1) {
	if (num < 1e3) return roundToDecimal(num, decimalCount);
	if (num < 1e6) return roundToDecimal(num / 1e3, decimalCount) + "k";
	if (num < 1e9) return roundToDecimal(num / 1e6 , decimalCount) + "m";
	if (num < 1e12) return roundToDecimal(num / 1e9, decimalCount) + "b";
	return roundToDecimal(num / 1e12, decimalCount) + "t";
}
```
you only need to call formatNumberForDisplay(planet.energy) for example

Reply:
resolved

Stx69 did not implement it, but its not a must, maybe he likes the exact numbers

Reply:
will looking on it.. I was bussy with multitasking...